eclair-core/src/test/scala/fr/acinq/eclair/channel/publish/ReplaceableTxPublisherSpec.scala

```diff
@@ -358,16 +409,196 @@ class ReplaceableTxPublisherSpec extends TestKitBaseClass with AnyFunSuiteLike w
       assert(result.tx.txIn.map(_.outPoint.txid).contains(commitTx.tx.txid))
       assert(result.tx.txIn.length > 2) // we added more than 1 wallet input
       assert(mempoolTxs.map(_.txid).contains(result.tx.txid))
-    })
+    }
+  }
+
+  test("commit tx fees not increased when confirmation target is far and feerate hasn't changed") {
+    withFixture(Seq(500 millibtc), ChannelTypes.AnchorOutputsZeroFeeHtlcTx) { f =>
+      import f._
+
+      val (commitTx, anchorTx) = closeChannelWithoutHtlcs(f, aliceBlockHeight() + 30)
+      wallet.publishTransaction(commitTx.tx).pipeTo(probe.ref)
+      probe.expectMsg(commitTx.tx.txid)
+
+      setFeerate(FeeratePerKw(3000 sat))
+      publisher ! Publish(probe.ref, anchorTx)
+      // wait for the commit tx and anchor tx to be published
+      val mempoolTxs = getMempoolTxs(2)
+      assert(mempoolTxs.map(_.txid).contains(commitTx.tx.txid))
+
+      // A new block is found, but we still have time and the feerate hasn't changed, so we don't bump the fees.
+      // Note that we don't generate blocks, so the transactions are still unconfirmed.
+      system.eventStream.publish(CurrentBlockCount(aliceBlockHeight().toLong + 5))
+      probe.expectNoMessage(500 millis)
+      val mempoolTxs2 = getMempool()
+      assert(mempoolTxs.map(_.txid).toSet === mempoolTxs2.map(_.txid).toSet)
+    }
+  }
+
+  test("commit tx not confirming, lowering anchor output amount") {
```

Comment:
I've got the following error message in this test (but the test still passes):

```
17:13:15.885 WARN  f.a.e.b.b.r.BitcoinCoreClient - cannot unlock utxo=UnlockOutpoint(3de35f812c58f093b956c6dc183649865c982573428a6fe488eea6232afeaf97,1):fr.acinq.eclair.blockchain.bitcoind.rpc.JsonRPCError: Invalid parameter, expected unspent output (code: -8)
	at fr.acinq.eclair.blockchain.bitcoind.rpc.BasicBitcoinJsonRPCClient.jsonResponse2Exception(BasicBitcoinJsonRPCClient.scala:56)
	at fr.acinq.eclair.blockchain.bitcoind.rpc.BasicBitcoinJsonRPCClient.$anonfun$invoke$1(BasicBitcoinJsonRPCClient.scala:53)
	at scala.concurrent.impl.Promise$Transformation.run(Promise.scala:431)
	at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1426)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
```

Pretty sure that's expected, but maybe we could handle that case so it's less messy and frightening.

Reply:
It's coming from the clean-up of the previous test, where I aggressively stop the actors (see the end of `withFixture`).
We can just remove this, and let the test harness destroy the actors, I'll try removing this clean-up code.

Reply:
I removed the test cleanup in https://github.com/ACINQ/eclair/pull/2113/commits/67e0d962f11ca024f2b1b84e9eecd7b74753b4b6, so it only appears after the tests that explicitly unlock early (where it's expected).